### ISSUE TITLE
🦠 Model Request: Tox21 ST-ATAD5

### ISSUE THREAD

**Issue by GemmaTuron** [ersilia-os/ersilia#435]
### Model Title

Tox21 ST-ATAD5 (TDC Dataset)

### Publication

Hello @mahamtariq!

As part of your Outreachy contribution, we have assigned you the dataset "Tox21 ST-ATAD5 (" from the Therapeutics Data Commons to try and build a binary classification ML model.
Please copy the provided Google Colab template and use this issue to provide updates on the progress. We'll value not only being able to build the model but also interpreting its results.

**Comment by mahamtariq58** [comment]
Thanks @GemmaTuron

**Comment by mahamtariq58** [comment]
### Data set Overview:

- [x] Upload the assigned dataset from TDC
- [x] Get the train/validation/test splits from TDC
- [x] Save the dataset splits in Google Drive as .csv

Here is the link to my [Google Colab Notebook](https://colab.research.google.com/drive/1_6i5VSKB23HO_GmvvHEJQLPunJFwi-Dn?usp=sharing)

**Comment by mahamtariq58** [comment]
### Data Analysis

- Tox21is a data challenge that includes qualitative measurement for approximately 10K compounds on 12  different assays
- The Aim of the Tox21 is to speed up toxicity testing while reducing the time and resources needed
- The goal is to efficiently test whether certain chemical interfere with the biological  processes that can lead to toxicity
- These toxic effects includes stress response effects(SR)  and nuclear receptor effects (NR). Both NR and SR effects are significantly relevant to human health, since activation of nuclear receptors
can disrupt endocrine system function (Chawla et al., 2001; Grün and Blumberg, 2007), and activation of stress response pathways can lead to liver injury or cancer (Bartkova et al., 2005; Labbe et al., 2008; Jaeschke et al., 2012)

### ATAD5 Receptor (ATAD5)
- It is cell-based assay using embroynic kidney cells was used to screen the Tox21compounds library. The assay uses a luciferase repoter-gene tagged with ATAD5 to measure the induction of ELG1 protein, and its increase marks a chemically induced genetic stress.

**1.**  What are we trying to predict?
We are predicting whether a drug activates a stress response pathway or not. If exposure to environmental chemicals and drugs activates the stress response in the cell the drug is labelled as toxic.

**Comment by GemmaTuron** [comment]
Good start @mahamtariq58 !

**Comment by mahamtariq58** [comment]
### Training ML Model:
For training the model, we need to segregate data into X_train, and Y_Train in order for the model to learn from it. We are using MorganBinaryClassifier to check the active and inactive molecules. I set the time_buget_sec to 30 min as my training data is imbalanced( more inactive than active molecules).After training the model, I saved the model in joblib format to load at any time.

**Comment by mahamtariq58** [comment]
### Model Evaluation 

Metrics for model evaluation for both Validation and Test Sets:

- **ROC Curve**
- **Confusion Matrix**
- **Precision and Recall**

_For Validation Set:_

**ROC Curve**
ROC curve shows the performance of the classification model at all classification thresholds. AUC (area under curve =0.78) represents the aggregate performance across all possible thresholds. The metrics show that the classifier is working quite well in labeling active and inactive molecules for the validation set.
![Roc_validation](https://user-images.githubusercontent.com/11984160/199336989-c8a8db86-2c24-4f47-8657-f0b558ba5840.png)

_For Test Set:_
**ROC Curve**
The figure shows the roc curve for the test data. As the ROC curve is more inclined towards 1, it represents that model performed efficiently in segregating active and inactive molecule.
![Roc_test](https://user-images.githubusercontent.com/11984160/199337015-f5f8a285-d8c2-4c9c-bb7b-f08528585c1f.png)

**Comment by mahamtariq58** [comment]
**Confusion Matrix**:
Confusion matrices show counts between expected and observed values. The result "TN" stands for True Negative and displays the number of negatively classed cases that were correctly identified. Similarly, "TP" stands for True Positive and denotes the quantity of correctly identified positive cases. 
_For Validation Set_ :
The matrix shows that out of (TP+TN=679) molecules, only (FP+FN=28) molecules were falsely predicted. The values indicate that model performed well.

![Confusion_Validation](https://user-images.githubusercontent.com/11984160/199341660-509967c7-3569-4d1b-a8e9-701a196ef914.png)

_For Test Set_:
Similarly, for the test set out of (TN+TP=1359) true labeled molecules, only (FP+FN=55) molecules were not correctly predicted.

![Confusion_test](https://user-images.githubusercontent.com/11984160/199341816-b27b85d6-55f4-4cba-87ca-d2e5ac724427.png)

**Comment by mahamtariq58** [comment]
**Precision & Recall:**

Precision is defined as 
![Screenshot from 2022-11-02 02-21-11](https://user-images.githubusercontent.com/11984160/199344146-64aca968-cb05-4b8c-aad8-55dc24b809c7.png)
Recall is defined as the
 
![new](https://user-images.githubusercontent.com/11984160/199345148-6ae70237-2cb0-4d1b-abbe-179fa4b3c833.png)

_For Validation Set:_
The precision value for the validation set is, which shows that our model predicted whether a molecule is active or inactive, it is correct 58% of the time. The Recall value shows that 23% of all active molecules are correctly classified by our model.

![Screenshot from 2022-11-02 02-24-21](https://user-images.githubusercontent.com/11984160/199344667-0dfe44d2-dcc6-4415-990c-98ef4452bf6b.png)

For Test Set:
For the test split, the precision value is 42%, which shows that the classifier correctly identifies the active and inactive molecules 42% of the time. The recall value depicts that 5.5% of all active molecules are correctly labeled.

![Screenshot from 2022-11-02 02-30-37](https://user-images.githubusercontent.com/11984160/199345668-59273963-520f-4b6e-980a-dffccad63d97.png)

The updated link of [Google Colab Notebook](https://colab.research.google.com/drive/1Jc378qCAzC-JjK8LPJ1gpRQ2BCIxLVko#scrollTo=MOCrY3kLWD5U)

**Comment by mahamtariq58** [comment]
Hi@GemmaTuron , This issue should close now or not?or do I need to add any changes?

**Comment by GemmaTuron** [comment]
Hello @mahamtariq58   !

Thanks for your contributions to Ersilia during the Outreachy Application Period! We hope you have learnt and enjoyed as much as the Ersilia team did!
I will close this issue as this was part of the application period solely.
